### ISSUE TITLE
Using pseudo protocol for log output

### ISSUE THREAD

**Issue by athoune** [containous/traefik#2151]
Traefik use two kinds of logs : system logs and access logs, with logrus.

For now, the options are tiny, json or flat, a file path.

Why don't use few output, using logrus plugins :

* syslog://
* journald://
* fluentd://
* logstash://
* file://

Something like Docker logging plugin, https://docs.docker.com/engine/admin/logging/overview/#use-environment-variables-or-labels-with-logging-drivers without the proprietary one.

Logrus output list huge. "Compile your own binary" like Caddy does is boring, and https://golang.org/pkg/plugin/ seems not ready.

`tail -f`, like filebeat and everyone use is hugly, and painful with containers.

What is the simplest answer for log routing ?

**Comment by tnucera** [comment]
any news for this feature ?

**Comment by fcatae** [comment]
+1 interested in this feature. 

I implemented a simple integration of AccessLog with Azure AppInsights (https://github.com/fcatae/traefik/commit/a727c22856d46f71d485cd3c9e051cddd9c570ab), and it seems to be easy to generalize for different logrus hooks.

//cc: @jjcollinge , @eladiw

**Comment by andrewthetechie** [comment]
+1 traefik logging to syslog would be a fantastic feature.

**Comment by avendretter** [comment]
+1 also here. Especially for running in a windows-based environment where you cant send SIGUSR1, no logrotate is possible.

**Comment by t0p4** [comment]
I join, I am also very interested in such functionality

**Comment by DrPyser** [comment]
Yes, it would be nice to avoid having to log to a file and send logs directly to some endpoint.

**Comment by Mhs-220** [comment]
This will be so cool, any news about it?

**Comment by papey** [comment]
A PR is waiting on #7439 and this issue is used to discuss the feature implemented in https://github.com/bearstech/traefik/tree/add-fluentd-draft.

For now, a quite basic setup with a `fluentd` container is working using a simple `--fluent` flag.

We need to discuss how all the things could be presented to end users.

**Comment by ad-m** [comment]
I have recently been working with Traefik & named FIFO to handle Traefik log processing. I would like to note that in case of blocking writes of access logs, Traefik may cease to be completely unresponsive. When writing to a regular file this is unlikely, but when writing to a named FIFO / TCP socket it is likely due lack of reader / network issue. Part of the problem can be mitigated by appropriate documentation, but I share my observations.

**Comment by athoune** [comment]
fluentd protocol handles network chaos, with buffering and dropping message when its queue is full. fluent logging is async by design, and your don't break the emmiter if the receiver is down. Mitigation is done in fluentd protocol design.

**Comment by papey** [comment]
For an overview of such features, head to https://docs.fluentd.org/deployment/failure-scenarios

**Comment by ad-m** [comment]
> fluentd protocol handles network chaos, with buffering and dropping message when its queue is full. fluent logging is async by design, and your don't break the emmiter if the receiver is down. Mitigation is done in fluentd protocol design.

My remark did not refer to Fluentd directly, but was broader in nature. I see that the issue can be mitigated with adequate documentation, as indicated in the case of Fluentd, pointing to the use of a local forwarder. The risk must be considered in detail in the context of a specific protocol.

**Comment by papey** [comment]
A WIP/Draft PR is available here https://github.com/bearstech/traefik/pull/1 to discuss implementation details, thanks !

**Comment by papey** [comment]
Hi :wave: 

Is it possible to restart the discussion on the implementation started here https://github.com/bearstech/traefik/pull/1 ?

(The MR was moved to an external repo in order to not explode the Traefik CI during the Hackathon)

Thanks a lot.
